### PR TITLE
Added Modular Exponential [Javascript]

### LINKED ISSUE #299: Modular Exponential [JavaScript]
#### This issue is about [Modular Exponential](https://en.wikipedia.org/wiki/Euclidean_algorithm).

- [x] Issue Title is in form `Algo/DS Name [Language]`.
- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [ ] This Algo/DS is worth adding to this repository.
- [x] Assign this issue to me (I want to work on this).
- [x] I will send my PR only after I'm assigned to this issue by any maintainer.

### PR DESCRIPTION
**Pull request by aashutoshrathi**
Fixes #299 

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.

### CODE REVIEW

File: modular_exponential/modular_exponential.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	:param power: Power to which the number is raised.
+	:param mod: Number by which modulo has to be performed.
+	:return: Returns (base ^ power) % mod
+	*/
+	let answer = 1;
+	base = base % mod;
+	while (power) {
+		if (power & 1) {
+			answer = (answer * base) % mod;
+	}
```
Comment (Monal5031) [inline modular_exponential/modular_exponential.js]: This bracket needs another indent here to align with `if ()`

File: modular_exponential/modular_exponential.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+			answer = (answer * base) % mod;
+		}
+		power = power >> 1;
+		base = (base * base) % mod;
+	}
+	return answer;
+}
+
+function main () {
+	let base = 2;
```
Comment (aviaryan) [inline modular_exponential/modular_exponential.js]: We should use a better example here that actually does some mod. eg (3 ^ 5 % 61)

File: modular_exponential/modular_exponential.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+function modularExponential (base, power, mod) {
```
Comment (aviaryan) [inline modular_exponential/modular_exponential.js]: The filename should be modularExponential.js

File: README.md [hunk 4]
```diff
@@ -30,7 +30,7 @@ Community (college) maintained list of Algorithms and Data Structures implementa
...
 | [Merge Sort](https://www.khanacademy.org/computing/computer-science/algorithms/merge-sort/a/overview-of-merge-sort) | [:white_check_mark:](merge_sort/merge_sort.c) | | [:white_check_mark:](merge_sort/MergeSort.java) | [:white_check_mark:](merge_sort/merge_sort.py) | | |
-| [Modular Exponential](http://www.geeksforgeeks.org/modular-exponentiation-power-in-modular-arithmetic/) | [:white_check_mark:](modular_exponential/modular_exponential.c) | | [:white_check_mark:](modular_exponential/ModularExponential.java) | [:white_check_mark:](modular_exponential/modular_exponential.py) | [:white_check_mark:](modular_exponential/modular_exponential.go) | |
+| [Modular Exponential](http://www.geeksforgeeks.org/modular-exponentiation-power-in-modular-arithmetic/) | [:white_check_mark:](modular_exponential/modular_exponential.c) | | [:white_check_mark:](modular_exponential/ModularExponential.java) | [:white_check_mark:](modular_exponential/modular_exponential.py) | [:white_check_mark:](modular_exponential/modular_exponential.go) | [:white_check_mark:](modular_exponential/modular_exponential.js) |
```
Comment (aviaryan) [inline README.md]: Please fix the link here.

**Pull request merged**